Fruitties/androidApp/src/main/java/com/example/fruitties/android/ui/ListScreen.kt

```diff
@@ -47,18 +47,38 @@ import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.MutableCreationExtras
 import androidx.lifecycle.viewmodel.compose.viewModel
 import com.example.fruitties.android.R
+import com.example.fruitties.android.di.App
 import com.example.fruitties.database.CartItemDetails
 import com.example.fruitties.model.Fruittie
+import com.example.fruitties.viewmodel.MainViewModel
 
 @Composable
-fun ListScreen(viewModel: MainViewModel = viewModel(factory = MainViewModel.Factory)) {
+fun ListScreen() {
+    // Instantiate a ViewModel with a dependency on the AppContainer.
+    // To make ViewModel compatible with KMP, the ViewModel factory must
+    // create an instance without referencing the Android Application.
+    // Here we put the KMP-compatible AppContainer into the extras
+    // so it can be passed to the ViewModel factory.
+    val app = LocalContext.current.applicationContext as App
+    val extras = remember(app) {
+        MutableCreationExtras().apply {
+            set(MainViewModel.Companion.AppContainerKey, app.container)
```

Comment:
Nit: the `Companion` can be omitted (`set(MainViewModel.AppContainerKey, app.container)`).

Reply:
It took a few steps to omit `Companion`.

I was using
`object AppContainerKey : CreationExtras.Key<AppContainer>`
which must be accessed with
`MainViewModel.Companion.AppContainerKey`

I changed my code to
`val AppContainerKey = object : CreationExtras.Key<AppContainer> {}`
which can be accessed with
`MainViewModel.AppContainerKey` or `MainViewModel.Companion.AppContainerKey`.

I found the syntax `object : Key<T> {}` in an inline function of `CreationExtras`.

`@JvmStatic public inline fun <reified T> Key(): Key<T> = object : Key<T> {}`

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:lifecycle/lifecycle-viewmodel/src/commonMain/kotlin/androidx/lifecycle/viewmodel/CreationExtras.kt;l=68?q=CreationExtras

However, the version of `CreationExtras` in release 2.8.4 is missing the companion object and some functions, so I cannot use the inline function Key() in the link above.

![Screenshot 2024-08-23 at 16 00 44](https://github.com/user-attachments/assets/33c3064f-d176-4fc1-b978-c1a05150121e)

If the inline function was available, then I could do this:
`val AppContainerKey = CreationExtras.Key<AppContainer>()`
which would be a little cleaner than
`val AppContainerKey = object : CreationExtras.Key<AppContainer> {}`